sklearn/feature_selection/_from_model.py

```diff
@@ -174,9 +192,16 @@ class SelectFromModel(MetaEstimatorMixin, SelectorMixin, BaseEstimator):
     array([False,  True, False])
     >>> selector.transform(X)
     array([[-1.34],
-           [-0.02],
-           [-0.48],
-           [ 1.48]])
+          [-0.02],
+          [-0.48],
+          [ 1.48]])
+
+    Using a callable to create a selector that can use no more than half
+    of the input features.
+
+    >>> half_selector = SelectFromModel(estimator = LogisticRegression())
+    >>> half_selector.set_params(max_features = lambda X: round(len(X[0]) / 2))
+    >>> half_selector.fit(X,y)
```

Comment:
We can set `max_features` directly, to avoid needing to introduce `set_params` here:

```suggestion
    >>> half_callable = lambda X: round(len(X[0]) / 2)
    >>> half_selector = SelectFromModel(estimator=LogisticRegression(),
    ...                                 max_features=half_callable)
    >>> _ = half_selector.fit(X, y)
    >>> half_selector.max_features_
    2
```

Also no spacing between the `=` is consistent with `black` or `flake8`. (Also to pass doctests)

**Side Note**: You can run doctest locally with

```bash
pytest sklearn/feature_selection/_from_model.py
```